app/src/org/commcare/tasks/ProcessAndSendTask.java

```diff
@@ -273,13 +273,10 @@ private boolean blockUntilTopOfQueue() throws TaskCancelledException {
 
     private void sendForms(FormRecord[] records) throws TaskCancelledException {
         for (int i = 0; i < records.length; ++i) {
-            //See whether we are OK to proceed based on the last form. We're now guaranteeing
-            //that forms are sent in order, so we won't proceed unless we succeed. We'll also permit
-            //proceeding if there was a local problem with a record, since we'll just move on from that
-            //processing.
-            if (i > 0 && !(results[i - 1] == FormUploadResult.FULL_SUCCESS || results[i - 1] == FormUploadResult.RECORD_FAILURE)) {
-                //Something went wrong with the last form, so we need to cancel this whole shebang
-                Logger.log(LogTypes.TYPE_WARNING_NETWORK, "Cancelling submission due to network errors. " + (i - 1) + " forms succesfully sent.");
+
+            if (previousFailurePredictsFutureFailures(results, i)) {

```

Comment:
I think this might be slightly misleading as a documenting-method-name. 

It's not that the previous failure implies the next form will fail, it's that submitting the previous form was mandatory for consistency reasons (IE: We couldn't send a create form, so we need to not try to submit the followup form)

Reply:
@ctsims That's not at all what this method checks for. Go look at its implementation (which is not something I wrote; I just refactored the code that was already there to be in its own method). All we have ever done here is to check what type of error caused the last submission to fail, and stop the process if it was an error type that indicates future submissions are also likely to fail for the same reason. There's nothing in our code that checks the type of the form that failed and employs any logic based on that.

Reply:
Hey, sorry, wasn't trying to make any assertions about type.

I was trying to say that the the name seems slightly off. The word "Predicts" implies that the reason we should stop submitting is because the next submission will fail, but the reason we should stop submitting is that the protocol requires that we send form N before we send form N+1 (the create/followup example was just meant to illustrate that, sorry for confusing.